pilot/cmd/pilot-discovery/main.go

```diff
@@ -102,6 +84,22 @@ var (
 	}
 )
 
+func initServerArgs(p *bootstrap.PilotArgs) {
```

Comment:
Could you pass this function directly to NewPilotArgs? That would require one less reference jump when searching for the default values from the call site of serverArgs/